Targets transition mega PR (#2)
USGS-R/ds-pipelines-targets-2-course: responses/32-file-targets.md

> File targets are very flexible and of course, are also easy to share or store elsewhere. 

Reviewer: ```suggestion
File targets are very flexible and, of course, are also easy to share or store elsewhere. 
```

> Additionally, many file targets are either language agnostic (e.g., csv, tsv, txt, nc files) _or_ are meant to be shared across languages, such as the how the [feather file](https://blog.rstudio.com/2016/03/29/feather/) was designed for exchange between R and Python.

Reviewer: ```suggestion
Additionally, many file formats are either language agnostic (e.g., csv, tsv, txt, nc files) _or_ are meant to be shared across languages, such as the [feather format](https://blog.rstudio.com/2016/03/29/feather/) designed for exchange between R and Python.
```

> When specifying a target in a makefile recipe with file targets, the path to the file needs to be either absolute or relative to the working directory that the `_targets.R` file is in. 

Reviewer: ```suggestion
When specifying a file target in a makefile, the path to the file needs to be either absolute or relative to the working directory that the `_targets.R` file is in. 
```

---

> Since file targets in the `targets` package are not the default and require you to add `format = "file"`, you may feel deterred from using files as targets. It's true, the benefits of files are quite small compared to the advantages of using objects. For reasons that will become clearer in the future, we recommend that files be used more liberally than objects because of two reasons: 1) ability to store data remotely in file format, and 2) ease of collaboration. You'll here more about this in the next pipelines courses and when you see some more of the team's pipelines in practice. 

Reviewer: ```suggestion
Since file targets in the `targets` package are not the default and require you to add `format = "file"`, you may feel deterred from using files as targets. It's true, the benefits of files are often small compared to the advantages of using objects. However, we still recommend that files be used liberally, especially for targets that you'll want to access outside of R (e.g., browsing figure files in Finder/Windows Explorer; opening a spatial data file in a GIS) or share with others (e.g., using outputs from one pipeline as inputs to another). 
```

Reviewer: I've changed this one pretty substantially and welcome your thoughts on our approach to file targets in `targets`. In the absence of a shared cache, I see a lot less need to push for file targets. And the remote storage argument doesn't quite fit because `targets` does offer support for [storing data on S3](https://books.ropensci.org/targets/cloud.html#storage), and besides, we're trying out more shared storage on Caldera these days. But there are still times when local file targets are handy - I added some in my suggested change.

Author: I like these suggestions; I think they give much less cryptic reasons for why you might choose a file target. I forget that we really liked file targets because of the shared cache dimension. That's a good point.
